src/components/GroupList.tsx

```diff
@@ -1,6 +1,103 @@
-
-import { Flex } from '@chakra-ui/react';
-
+import {
+  Container,
+  Box,
+  Flex,
+  useTheme,
+  Avatar,
+  Heading,
+  Text,
+  Stack,
+  StackDivider,
+  Image,
+} from "@chakra-ui/react";
+import netflixImage from "./images/netflix.png";
+import spotify from "./images/spotify.png";
 export default function GroupList() {
-    return <Flex>This show show a list of groups.</Flex>
-}
\ No newline at end of file
+  const theme = useTheme();
+  const subscriptionCost = "$8.10";
+  const savings = "$14.90";
+  const profilePicture = "https://bit.ly/sage-adebayo";
+  const subscriptions = [
+    {
+      name: "Netflix",
+      image: netflixImage,
+    },
+    {
+      name: "Spotify",
+      image: spotify,
+    },
+  ];
+  return (
+    <Container bg={theme.colors.secondary[600]}>
+      <Box
+        className="Header"
+        bg={theme.colors.primary[500]}
+        w="100%"
+        p={4}
+        color="white"
+      >
+        <Heading as="h1" size="1xl" textAlign="center">
+          Groups
+        </Heading>
+      </Box>
+      <Flex className="Profile" margin="10px">
+        <Avatar src={profilePicture} marginRight="10px" />
+        <Box>
+          <Text fontWeight="bold"> Total Subscriptions</Text>
+          <Text>
+            You owe{" "}
+            <Text as="span" fontWeight="bold">
+              {subscriptionCost}
+            </Text>
+            /month
+          </Text>
+          <Text fontWeight="bold" color="green">
+            {savings}
+            <Text as="span" fontSize="xs" color="black" fontWeight="thin">
+              {" "}
+              Saved
+            </Text>
+          </Text>
+        </Box>
+      </Flex>
+      <Stack
+        className="MyGroup"
+        direction={"column"}
+        bg="white"
+        divider={<StackDivider borderColor="gray.200" />}
+        spacing={4}
+        align="stretch"
+      >
+        <Text fontWeight="bold">My Groups</Text>
+        <Box>
+          {" "}
+          <Flex className="profile" margin="10px">
+            <Image
+              src={subscriptions[0].image}
+              alt={subscriptions[0].name}
+              marginRight="10px"
+            />
+            <Box>
```

Comment:
This entire section can be a separate component that takes in a prop.

For example
```
function Subscription(props: { cost: string, image: string, name: string }) {
   return <Box>
       ...
    </Box>
}
```

In the component above, you can use src={props.image} instead of src={subscriptions[0].image}.
To use the new component:
```
{
subscriptions.map(subscription => <Subscription image={subscription.image} cost={'the cost'} name={subscription.name}/>)
}
```

That way we can use the same subscription instead of having to copy and paste

Reply:
made a commit with changes for more reusable code